core/trino-main/src/test/java/io/trino/security/TestFileBasedSystemAccessControl.java

```diff
@@ -114,6 +114,28 @@ public void testCanImpersonateUserOperations()
         catch (AccessDeniedException expected) {
         }
 
+        accessControlManager.checkCanImpersonateUser(Identity.ofUser("svc_tenant"), "svc_tenant_prod");
+        try {
+            accessControlManager.checkCanImpersonateUser(Identity.ofUser("svc_tenant"), "svc_tenant_other");
+            throw new AssertionError("expected AccessDeniedException");
+        }
+        catch (AccessDeniedException expected) {
+        }
+        try {
+            accessControlManager.checkCanImpersonateUser(Identity.ofUser("svc_tenant"), "svc_other_prod");
+            throw new AssertionError("expected AccessDeniedException");
+        }
+        catch (AccessDeniedException expected) {
+        }
+
+        accessControlManager.checkCanImpersonateUser(Identity.ofUser("external_corp_dept"), "internal-corp-dept-sandbox");
+        try {
+            accessControlManager.checkCanImpersonateUser(Identity.ofUser("external_corp_dept"), "invalid");
+            throw new AssertionError("expected AccessDeniedException");
+        }
+        catch (AccessDeniedException expected) {
+        }
```

Comment:
Could you rebase on upstream and change to use `assertThatThrownBy` instead?

Reply:
Done.